<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>CollectionRebuilder</class>
 <widget class="QDialog" name="CollectionRebuilder">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>498</width>
    <height>470</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>ROM Collection Rebuilder</string>
  </property>
  <layout class="QGridLayout" name="gridLayout">
   <item row="0" column="0">
    <widget class="QLabel" name="labelXmlSource">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="text">
      <string>XML source</string>
     </property>
    </widget>
   </item>
   <item row="0" column="1">
    <widget class="QComboBox" name="comboBoxXmlSource">
     <property name="sizePolicy">
      <sizepolicy hsizetype="MinimumExpanding" vsizetype="Minimum">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="toolTip">
      <string>Select XML source</string>
     </property>
     <item>
      <property name="text">
       <string>Current default emulator</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Select XML file...</string>
      </property>
     </item>
    </widget>
   </item>
   <item row="3" column="0" colspan="3">
    <widget class="QFrame" name="frameEntities">
     <layout class="QGridLayout" name="gridLayoutFrameEnitities">
      <property name="margin">
       <number>0</number>
      </property>
      <item row="1" column="1">
       <widget class="QLineEdit" name="lineEditSetEntity">
        <property name="toolTip">
         <string>Entity name that defines a 'set'</string>
        </property>
       </widget>
      </item>
      <item row="1" column="3">
       <widget class="QLineEdit" name="lineEditRomEntity">
        <property name="toolTip">
         <string>Entity name that defines a 'rom'</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="labelSetEntity">
        <property name="text">
         <string>Set entity</string>
        </property>
       </widget>
      </item>
      <item row="1" column="4">
       <widget class="QLabel" name="labelDiskEntity">
        <property name="toolTip">
         <string>Entity name that defines a 'disk'</string>
        </property>
        <property name="text">
         <string>Disk entity</string>
        </property>
       </widget>
      </item>
      <item row="1" column="5">
       <widget class="QLineEdit" name="lineEditDiskEntity">
        <property name="toolTip">
         <string>Entity name that defines a 'disk'</string>
        </property>
       </widget>
      </item>
      <item row="1" column="2">
       <widget class="QLabel" name="labelRomEntity">
        <property name="text">
         <string>ROM entity</string>
        </property>
       </widget>
      </item>
      <item row="0" column="0" colspan="6">
       <widget class="Line" name="line2">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="4" column="0" colspan="3">
    <widget class="QPlainTextEdit" name="plainTextEditLog">
     <property name="verticalScrollBarPolicy">
      <enum>Qt::ScrollBarAlwaysOn</enum>
     </property>
     <property name="horizontalScrollBarPolicy">
      <enum>Qt::ScrollBarAlwaysOn</enum>
     </property>
     <property name="lineWrapMode">
      <enum>QPlainTextEdit::NoWrap</enum>
     </property>
     <property name="readOnly">
      <bool>true</bool>
     </property>
    </widget>
   </item>
   <item row="6" column="0" colspan="3">
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <widget class="QCheckBox" name="checkBoxEnableLog">
       <property name="toolTip">
        <string>Enable log output</string>
       </property>
       <property name="text">
        <string>Enable log</string>
       </property>
       <property name="checked">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item>
      <widget class="Line" name="line">
       <property name="orientation">
        <enum>Qt::Vertical</enum>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="labelMaxLogSize">
       <property name="text">
        <string>Log size</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QSpinBox" name="spinBoxMaxLogSize">
       <property name="toolTip">
        <string>Maximum number of lines in log (0 = no limit)</string>
       </property>
       <property name="specialValueText">
        <string>unlimited</string>
       </property>
       <property name="suffix">
        <string> lines</string>
       </property>
       <property name="minimum">
        <number>0</number>
       </property>
       <property name="maximum">
        <number>999999</number>
       </property>
       <property name="singleStep">
        <number>100</number>
       </property>
       <property name="value">
        <number>0</number>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QProgressBar" name="progressBar">
       <property name="toolTip">
        <string>Rebuild progress indicator</string>
       </property>
       <property name="value">
        <number>0</number>
       </property>
       <property name="format">
        <string>Idle</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QToolButton" name="toolButtonClear">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Fixed" vsizetype="Minimum">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="toolTip">
        <string>Clear log</string>
       </property>
       <property name="text">
        <string>Clear</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QToolButton" name="toolButtonClose">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Fixed" vsizetype="Minimum">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="toolTip">
        <string>Close this window</string>
       </property>
       <property name="text">
        <string>Close</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item row="0" column="2">
    <layout class="QHBoxLayout" name="horizontalLayout_2">
     <item>
      <widget class="QToolButton" name="toolButtonRemoveXmlSource">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Fixed" vsizetype="Minimum">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="toolTip">
        <string>Remove the current XML source from the list</string>
       </property>
       <property name="icon">
        <iconset resource="qmc2.qrc">
         <normaloff>:/data/img/broom.png</normaloff>:/data/img/broom.png</iconset>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="pushButtonStartStop">
       <property name="sizePolicy">
        <sizepolicy hsizetype="MinimumExpanding" vsizetype="Minimum">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="toolTip">
        <string>Start / stop rebuilding</string>
       </property>
       <property name="text">
        <string>Start rebuilding</string>
       </property>
       <property name="icon">
        <iconset resource="qmc2.qrc">
         <normaloff>:/data/img/refresh.png</normaloff>:/data/img/refresh.png</iconset>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="pushButtonPauseResume">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="toolTip">
        <string>Pause / resume rebuilding</string>
       </property>
       <property name="text">
        <string>Pause</string>
       </property>
       <property name="icon">
        <iconset resource="qmc2.qrc">
         <normaloff>:/data/img/time.png</normaloff>:/data/img/time.png</iconset>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item row="5" column="0" colspan="3">
    <widget class="QFrame" name="frameRebuildStatus">
     <property name="frameShape">
      <enum>QFrame::StyledPanel</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Sunken</enum>
     </property>
     <layout class="QGridLayout" name="gridLayoutFrameRebuildStatus">
      <property name="margin">
       <number>0</number>
      </property>
      <item row="0" column="0">
       <widget class="QLabel" name="labelRebuildStatus">
        <property name="toolTip">
         <string>Rebuild statistics</string>
        </property>
        <property name="text">
         <string notr="true">&lt;center&gt;&lt;table border=&quot;0&quot; cellpadding=&quot;0&quot; cellspacing=&quot;4&quot;&gt;&lt;tr&gt;
&lt;td nowrap width=&quot;16.5%&quot; align=&quot;left&quot;&gt;&lt;b&gt;Sets processed:&lt;/b&gt;&lt;/td&gt;&lt;td nowrap width=&quot;16.5%&quot; align=&quot;right&quot;&gt;0&lt;/td&gt;
&lt;td nowrap align=&quot;center&quot;&gt;|&lt;/td&gt;
&lt;td nowrap width=&quot;16.5%&quot; align=&quot;left&quot;&gt;&lt;b&gt;Missing ROMs:&lt;/b&gt;&lt;/td&gt;&lt;td nowrap width=&quot;16.5%&quot; align=&quot;right&quot;&gt;0&lt;/td&gt;
&lt;td nowrap align=&quot;center&quot;&gt;|&lt;/td&gt;
&lt;td nowrap width=&quot;16.5%&quot; align=&quot;left&quot;&gt;&lt;b&gt;Missing disks:&lt;/b&gt;&lt;/td&gt;&lt;td nowrap width=&quot;16.5%&quot; align=&quot;right&quot;&gt;0&lt;/td&gt;
&lt;/tr&gt;&lt;/table&gt;&lt;/center&gt;</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="1" column="0" colspan="3">
    <widget class="Line" name="line1">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
   </item>
   <item row="2" column="0" colspan="3">
    <layout class="QGridLayout" name="gridLayoutFilters">
     <item row="0" column="0">
      <widget class="QCheckBox" name="checkBoxFilterExpressionSoftwareLists">
       <property name="toolTip">
        <string>Enable the use of a Qt regular expression that a software-list's name has to match in order to be processed</string>
       </property>
       <property name="text">
        <string>Filter lists</string>
       </property>
      </widget>
     </item>
     <item row="0" column="1">
      <widget class="QComboBox" name="comboBoxFilterSyntaxSoftwareLists">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="sizePolicy">
        <sizepolicy hsizetype="Preferred" vsizetype="Minimum">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="toolTip">
        <string>Choose the regular expression syntax</string>
       </property>
       <item>
        <property name="text">
         <string notr="true">RegExp</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string notr="true">RegExp2</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string notr="true">Wildcard</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string notr="true">WildcardUnix</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string notr="true">FixedString</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string notr="true">W3CXmlSchema11</string>
        </property>
       </item>
      </widget>
     </item>
     <item row="0" column="2">
      <widget class="QComboBox" name="comboBoxFilterTypeSoftwareLists">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="sizePolicy">
        <sizepolicy hsizetype="Preferred" vsizetype="Minimum">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="toolTip">
        <string>Select the filter type (include / exclude filter)</string>
       </property>
       <item>
        <property name="text">
         <string>Include</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Exclude</string>
        </property>
       </item>
      </widget>
     </item>
     <item row="0" column="3">
      <widget class="QLineEdit" name="lineEditFilterExpressionSoftwareLists">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="sizePolicy">
        <sizepolicy hsizetype="Expanding" vsizetype="Minimum">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="toolTip">
        <string>Specify a Qt regular expression that a software-list's name has to match in order to be processed</string>
       </property>
      </widget>
     </item>
     <item row="0" column="4">
      <widget class="QToolButton" name="toolButtonClearFilterExpressionSoftwareLists">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="sizePolicy">
        <sizepolicy hsizetype="Fixed" vsizetype="Minimum">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="toolTip">
        <string>Clear filter expression</string>
       </property>
       <property name="icon">
        <iconset resource="qmc2.qrc">
         <normaloff>:/data/img/broom.png</normaloff>:/data/img/broom.png</iconset>
       </property>
      </widget>
     </item>
     <item row="1" column="0">
      <widget class="QCheckBox" name="checkBoxFilterExpression">
       <property name="toolTip">
        <string>Enable the use of a Qt regular expression that a set's ID has to match in order to be processed</string>
       </property>
       <property name="text">
        <string>Filter sets</string>
       </property>
      </widget>
     </item>
     <item row="1" column="1">
      <widget class="QComboBox" name="comboBoxFilterSyntax">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="sizePolicy">
        <sizepolicy hsizetype="Preferred" vsizetype="Minimum">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="toolTip">
        <string>Choose the regular expression syntax</string>
       </property>
       <item>
        <property name="text">
         <string notr="true">RegExp</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string notr="true">RegExp2</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string notr="true">Wildcard</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string notr="true">WildcardUnix</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string notr="true">FixedString</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string notr="true">W3CXmlSchema11</string>
        </property>
       </item>
      </widget>
     </item>
     <item row="1" column="2">
      <widget class="QComboBox" name="comboBoxFilterType">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="sizePolicy">
        <sizepolicy hsizetype="Preferred" vsizetype="Minimum">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="toolTip">
        <string>Select the filter type (include / exclude filter)</string>
       </property>
       <item>
        <property name="text">
         <string>Include</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Exclude</string>
        </property>
       </item>
      </widget>
     </item>
     <item row="1" column="3">
      <widget class="QLineEdit" name="lineEditFilterExpression">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="sizePolicy">
        <sizepolicy hsizetype="Expanding" vsizetype="Minimum">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="toolTip">
        <string>Specify a Qt regular expression that a set's ID has to match in order to be processed</string>
       </property>
      </widget>
     </item>
     <item row="1" column="4">
      <widget class="QToolButton" name="toolButtonClearFilterExpression">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="sizePolicy">
        <sizepolicy hsizetype="Fixed" vsizetype="Minimum">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="toolTip">
        <string>Clear filter expression</string>
       </property>
       <property name="icon">
        <iconset resource="qmc2.qrc">
         <normaloff>:/data/img/broom.png</normaloff>:/data/img/broom.png</iconset>
       </property>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <tabstops>
  <tabstop>comboBoxXmlSource</tabstop>
  <tabstop>toolButtonRemoveXmlSource</tabstop>
  <tabstop>pushButtonStartStop</tabstop>
  <tabstop>pushButtonPauseResume</tabstop>
  <tabstop>checkBoxFilterExpressionSoftwareLists</tabstop>
  <tabstop>comboBoxFilterSyntaxSoftwareLists</tabstop>
  <tabstop>comboBoxFilterTypeSoftwareLists</tabstop>
  <tabstop>lineEditFilterExpressionSoftwareLists</tabstop>
  <tabstop>toolButtonClearFilterExpressionSoftwareLists</tabstop>
  <tabstop>checkBoxFilterExpression</tabstop>
  <tabstop>comboBoxFilterSyntax</tabstop>
  <tabstop>comboBoxFilterType</tabstop>
  <tabstop>lineEditFilterExpression</tabstop>
  <tabstop>toolButtonClearFilterExpression</tabstop>
  <tabstop>lineEditSetEntity</tabstop>
  <tabstop>lineEditRomEntity</tabstop>
  <tabstop>lineEditDiskEntity</tabstop>
  <tabstop>plainTextEditLog</tabstop>
  <tabstop>checkBoxEnableLog</tabstop>
  <tabstop>spinBoxMaxLogSize</tabstop>
  <tabstop>toolButtonClear</tabstop>
  <tabstop>toolButtonClose</tabstop>
 </tabstops>
 <resources>
  <include location="qmc2.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>toolButtonClose</sender>
   <signal>clicked()</signal>
   <receiver>CollectionRebuilder</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>492</x>
     <y>464</y>
    </hint>
    <hint type="destinationlabel">
     <x>431</x>
     <y>472</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>toolButtonClear</sender>
   <signal>clicked()</signal>
   <receiver>plainTextEditLog</receiver>
   <slot>clear()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>445</x>
     <y>464</y>
    </hint>
    <hint type="destinationlabel">
     <x>350</x>
     <y>329</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>toolButtonClearFilterExpression</sender>
   <signal>clicked()</signal>
   <receiver>lineEditFilterExpression</receiver>
   <slot>clear()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>478</x>
     <y>85</y>
    </hint>
    <hint type="destinationlabel">
     <x>424</x>
     <y>84</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>toolButtonClearFilterExpressionSoftwareLists</sender>
   <signal>clicked()</signal>
   <receiver>lineEditFilterExpressionSoftwareLists</receiver>
   <slot>clear()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>482</x>
     <y>56</y>
    </hint>
    <hint type="destinationlabel">
     <x>419</x>
     <y>56</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>checkBoxFilterExpressionSoftwareLists</sender>
   <signal>toggled(bool)</signal>
   <receiver>comboBoxFilterSyntaxSoftwareLists</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>57</x>
     <y>54</y>
    </hint>
    <hint type="destinationlabel">
     <x>122</x>
     <y>53</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>checkBoxFilterExpressionSoftwareLists</sender>
   <signal>toggled(bool)</signal>
   <receiver>comboBoxFilterTypeSoftwareLists</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>49</x>
     <y>60</y>
    </hint>
    <hint type="destinationlabel">
     <x>224</x>
     <y>60</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>checkBoxFilterExpressionSoftwareLists</sender>
   <signal>toggled(bool)</signal>
   <receiver>lineEditFilterExpressionSoftwareLists</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>56</x>
     <y>55</y>
    </hint>
    <hint type="destinationlabel">
     <x>299</x>
     <y>51</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>checkBoxFilterExpressionSoftwareLists</sender>
   <signal>toggled(bool)</signal>
   <receiver>toolButtonClearFilterExpressionSoftwareLists</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>38</x>
     <y>53</y>
    </hint>
    <hint type="destinationlabel">
     <x>479</x>
     <y>49</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>checkBoxFilterExpression</sender>
   <signal>toggled(bool)</signal>
   <receiver>comboBoxFilterSyntax</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>37</x>
     <y>85</y>
    </hint>
    <hint type="destinationlabel">
     <x>106</x>
     <y>85</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>checkBoxFilterExpression</sender>
   <signal>toggled(bool)</signal>
   <receiver>comboBoxFilterType</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>48</x>
     <y>86</y>
    </hint>
    <hint type="destinationlabel">
     <x>209</x>
     <y>85</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>checkBoxFilterExpression</sender>
   <signal>toggled(bool)</signal>
   <receiver>lineEditFilterExpression</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>33</x>
     <y>80</y>
    </hint>
    <hint type="destinationlabel">
     <x>347</x>
     <y>89</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>checkBoxFilterExpression</sender>
   <signal>toggled(bool)</signal>
   <receiver>toolButtonClearFilterExpression</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>22</x>
     <y>85</y>
    </hint>
    <hint type="destinationlabel">
     <x>479</x>
     <y>89</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
